spidermon/contrib/scrapy/monitors.py

```diff
@@ -0,0 +1,145 @@
+            item_extracted, minimum_threshold)
+        self.assertTrue(
+            item_extracted >= minimum_threshold, msg=msg
+        )
+
+
+@monitors.name('Log Monitor')
+class LogMonitor(BaseScrapyMonitor):
+    """ Check for errors in the spider log you can configure the expected
+    number of ERROR log messages using ``SPIDERMON_MAX_ERROR``,
+    the default is ``0``."""
+
+    @monitors.name('Should not have any errors')
+    def test_should_not_have_errors(self):
+        errors_threshold = self.get_settings(SPIDERMON_MAX_ERROR, 0)
+        no_of_errors = self.stats.get('log_count/ERROR', 0)
+        msg = 'Found {} errors in log, maximum expected is '\
+              '{}'.format(no_of_errors, errors_threshold)
+        self.assertTrue(no_of_errors <= errors_threshold, msg=msg)
+
+
+@monitors.name('Finish Reason Monitor')
+class FinishReasonMonitor(BaseScrapyMonitor):
+    """ Check if a job has a expected finish reason.
+    You can configure the expected reason with the
+    ``SPIDERMON_EXPECTED_FINISH_REASONS``, it should be an ``iterable`` of
+    valid finish reasons.
+
+    The default value of this settings is: ``['finished', ]``"""
+
+    @monitors.name('Should have the expected finished reason(s)')
+    def test_should_not_have_errors(self):
+        expected_reasons = self.get_settings(
+            SPIDERMON_EXPECTED_FINISH_REASONS, ('finished', ))
+        finished_reason = self.stats.get('finish_reason')
+        msg = 'Finished with "{}" the expected reasons are {}'.format(
+                finished_reason, expected_reasons)
+        self.assertTrue(finished_reason in expected_reasons, msg=msg)
+
+
+@monitors.name('Check for unwanted http status codes')
+class UnwantedHttpStatus(BaseScrapyMonitor):
+    """ Check for maximum number of unwanted http codes.
+
+    You can configure a list of unwanted http codes with
+    ``SPIDERMON_UNWANTED_HTTP_CODES`` the default value is:
+    ``[400, 407, 429, 500, 502, 503, 504, 523, 540, 541]``
+
+    You can also configured the **maximum** number of times that those
+    response codes with ``SPIDERMON_MAX_UNWANTED_HTTP_CODES`` the default
+    value is ``1``.
+    """
+    default_codes = [400, 407, 429, 500, 502, 503, 504, 523, 540, 541]
+
+    @monitors.name('Should not hit the limit of unwanted http status')
+    def test_check_unwanted_http_codes(self):
+        max_errors = self.get_settings(SPIDERMON_MAX_UNWANTED_HTTP_CODES, 1)
+        error_codes = self.get_settings(SPIDERMON_UNWANTED_HTTP_CODES,
+                                        self.default_codes)
+        for code in error_codes:
+            times = self.stats.get(
+                'downloader/response_status_count/{}'.format(code), 0)
+            msg = 'Found {} Responses with status code={} - '\
+                  'This exceed the limit of {}'.format(times, code, max_errors)
+            self.assertTrue(times < max_errors, msg=msg)
+
+
+class SpiderCloseMonitorSuite(MonitorSuite):
```

Comment:
Do we need to provide a default suite or only the basic monitors?
I ask because of the CLI client that, probably, will create this suite.

Reply:
This can be discussed further, but I think it's a **nice** approach to have more 5 lines her, this way the developer doesn't need to write this line in his own code nor the CLI needs to generate it :)

Reply:
Not sure about that, but if we do, we will need more info in the docs explaining how to enable it.

Reply:
There's documentation about how to enable this Suite, you think the documentation isn't clear or you didn't found it @rennerocha ?

Reply:
@raphapassini I found the docs now :-)